src/utils/url.js

```diff
@@ -92,64 +94,32 @@ export const tryParseUrl = (url: string, base?: string | URL): URL | void => {
 };
 
 /**
- * Test if the given URL string resolves on the realm, with realm as base.
- *
- * DEPRECATED because URL strings are complicated.  Callers should construct
- *   a URL object before this point, and use the same URL object both for
- *   asking any questions like this one and for subsequently making any
- *   network requests.
- *
- * This returns true just if `new URL(url, realm)` gives a URL that is
- * within the same Zulip realm.
- *
- * This performs a call to `new URL` and therefore may take a fraction of a
- * millisecond.  Avoid using in a context where it might be called more than
- * 10 or 100 times per user action.
+ * Test if the given URL is within the given realm.
  */
-// TODO: Take a URL object instead of a string, and remove warnings in jsdoc.
-export const isUrlOnRealm = (url: string, realm: URL): boolean => {
-  const parsedUrl = tryParseUrl(url, realm);
-  return parsedUrl ? parsedUrl.origin === realm.origin : false;
-};
+export const isUrlOnRealm = (url: URL, realm: URL): boolean => url.origin === realm.origin;
 
-const getResourceWithAuth = (uri: string, auth: Auth) => ({
-  uri: new URL(uri, auth.realm).toString(),
+const getResourceWithAuth = (url: URL, auth: Auth) => ({
+  uri: url.toString(),
   headers: getAuthHeaders(auth),
 });
 
-const getResourceNoAuth = (uri: string) => ({
-  uri,
+const getResourceNoAuth = (url: URL) => ({
+  uri: url.toString(),
```

Comment:
url.js [nfc]: Use "url", not "uri", in names we choose
    
    See Greg's findings about "url" being preferable to "uri":
      https://chat.zulip.org/#narrow/stream/92-learning/topic/URLs/near/794851
    
I decided to take this as a prompt to go write that up for our style guide:
https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#url-not-uri

So you can link to there instead.

(Also a nit which I found while writing that: the linked message is about "relative URL reference" vs. "URL", not about "URI". I think you meant the message before it.)